apps/predictions/lib/parser.ex

```diff
@@ -13,30 +14,83 @@ defmodule Predictions.Parser do
           0 | 1,
           DateTime.t() | nil,
           non_neg_integer,
-          Prediction.schedule_relationship(),
+          Prediction.schedule_relationship() | nil,
           String.t() | nil,
           String.t() | nil,
           boolean
         }
 
-  @spec parse(JsonApi.Item.t()) :: record
-  def parse(%JsonApi.Item{attributes: attributes} = item) do
+  @spec parse(Item.t()) :: record
+  def parse(%Item{} = item) do
     {
       item.id,
       trip_id(item),
       stop_id(item),
       route_id(item),
-      attributes["direction_id"],
-      [attributes["arrival_time"], attributes["departure_time"]] |> first_time,
-      attributes["stop_sequence"] || 0,
-      schedule_relationship(attributes["schedule_relationship"]),
+      direction_id(item),
+      first_time(item),
+      stop_sequence(item),
+      schedule_relationship(item),
       track(item),
-      attributes["status"],
-      departing?(attributes)
+      status(item),
+      departing?(item)
     }
   end
 
-  defp first_time(times) do
+  @spec departing?(Item.t()) :: boolean()
+  def departing?(%Item{attributes: %{"departure_time" => binary}}) when is_binary(binary),
+    do: true
+
+  def departing?(%Item{attributes: %{"status" => binary}}) when is_binary(binary),
+    do: upcoming_status?(binary)
```

Comment:
**question:** could you speak a bit to how the logic in `upcoming_status?` related to whether or not a prediction corresponds to a departure? It looks like it might have something to do with Commuter Rail, but then `upcoming_status?` also checks for the string "away" in the status which looks to me like it might be related to "stops away" on the rapid transit?

Reply:
That was added six years ago in https://github.com/mbta/dotcom-legacy/pull/1004/ :D 
It look like our current code used it as a way to guess whether a prediction corresponds to a departure, for predictions which have no departure time.
Do you suppose it might no longer be relevant?

Reply:
Ahh okay, I think that the "away" part at least must date from a different era when we didn't yet have minute countdowns on the Green Line but instead displayed [only the number of stops away](https://upload.wikimedia.org/wikipedia/commons/1/1f/Stops_away_countdown_signs_at_Park_Street%2C_January_2016.JPG). So that is likely no longer relevant. And that sign also shows "Approaching" and "Boarding."

Reply:
So I think from a subway perspective we can probably ditch this logic. I would just want to make sure there are no unintended consequences with Commuter Rail data since we also use the `status` field there.

Reply:
Also this PR has already got a lot of stuff going on and this logic isn't actually being changed as part of it, so I think it would be fine if you punted this to a tech debt task.